tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/model/tree/TmfTreeModel.java

```diff
@@ -35,6 +35,12 @@ public class TmfTreeModel<T extends ITmfTreeDataModel> {
     private List<ITableColumnDescriptor> fColumnDescriptors;
     private List<T> fEntries;
     private @Nullable String fScope;
+    /*
+     * Indicates which level of the tree should be expanded. If expandLevel is missing or has value of -1 it
+     * would mean that all the levels are expanded (default behaviour). If the expandLevel is set to 1 for
+     * example, it will expand to the first nodes under the root.
+     */
+    private int fExpandLevel = -1;
```

Comment:
There should be a constant available for -1 and used in the both classes. 

Add this to this class: 
```java
    /**
     * Expand level to indicate that all tree levels are expanded. (default)
     * @since 9.6
     */
    public static final int ALL_LEVELS = -1;
```

Also, rename variable to `fAutoExpandLevel`

Reply:
Done